[Bug] The maven version required to compile the project is too low, causing the compilation to fail

### Search before asking

- [x] I searched in the [issues](https://github.com/apache/paimon/issues) and found nothing similar.


### Paimon version

1.1-SNAPSHOT

### Compute Engine

No

### Minimal reproduce step

1. git clone git@github.com:apache/paimon.git
2. cd paimon
3. mvn clean install -DskipTests


### What doesn't meet your expectations?

```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Paimon : 1.1-SNAPSHOT .............................. SUCCESS [  6.459 s]
[INFO] Paimon : Test utils ................................ FAILURE [  7.140 s]
[INFO] Paimon : Common .................................... SKIPPED
[INFO] Paimon : Benchmark ................................. SKIPPED
[INFO] Paimon : Benchmark : Cluster ....................... SKIPPED
[INFO] Paimon : Code Gen .................................. SKIPPED
[INFO] Paimon : Code Gen Loader ........................... SKIPPED
[INFO] Paimon : Format .................................... SKIPPED
[INFO] Paimon : Core ...................................... SKIPPED
[INFO] Paimon : FileSystems : ............................. SKIPPED
[INFO] Paimon : FileSystems : S3 : Impl ................... SKIPPED
[INFO] Paimon : FileSystems : S3 .......................... SKIPPED
[INFO] Paimon : Hive ...................................... SKIPPED
[INFO] Paimon : Hive Common ............................... SKIPPED
[INFO] Paimon : Hive Catalog .............................. SKIPPED
[INFO] Paimon : Bundle .................................... SKIPPED
[INFO] Paimon : Benchmark : Micro ......................... SKIPPED
[INFO] Paimon : Service ................................... SKIPPED
[INFO] Paimon : Service : Client .......................... SKIPPED
[INFO] Paimon : Service : Runtime ......................... SKIPPED
[INFO] Paimon : Flink ..................................... SKIPPED
[INFO] Paimon : Flink : Common ............................ SKIPPED
[INFO] Paimon : Flink : CDC ............................... SKIPPED
[INFO] Paimon : Flink : 1.15 .............................. SKIPPED
[INFO] Paimon : Flink : 1.16 .............................. SKIPPED
[INFO] Paimon : Flink : 1.17 .............................. SKIPPED
[INFO] Paimon : Flink : 1.18 .............................. SKIPPED
[INFO] Paimon : Flink : 1.19 .............................. SKIPPED
[INFO] Paimon : Flink : 1.20 .............................. SKIPPED
[INFO] Paimon : Flink : Action ............................ SKIPPED
[INFO] Paimon : Spark ..................................... SKIPPED
[INFO] Paimon : Spark : Common : 2.12 ..................... SKIPPED
[INFO] Paimon : Docs ...................................... SKIPPED
[INFO] Paimon : Hive Connector Common ..................... SKIPPED
[INFO] Paimon : Spark3 : Common ........................... SKIPPED
[INFO] Paimon : Spark : UT ................................ SKIPPED
[INFO] Paimon : Spark : 3.3 ............................... SKIPPED
[INFO] Paimon : Tools : CI : Java ......................... SKIPPED
[INFO] Paimon : End to End Tests .......................... SKIPPED
[INFO] Paimon : FileSystems : Hadoop Shaded ............... SKIPPED
[INFO] Paimon : FileSystems : OSS : Impl .................. SKIPPED
[INFO] Paimon : FileSystems : OSS ......................... SKIPPED
[INFO] Paimon : Hive Connector : 2.1 ...................... SKIPPED
[INFO] Paimon : Hive Connector : 2.1-cdh-6.3 .............. SKIPPED
[INFO] Paimon : Hive Connector : 2.2 ...................... SKIPPED
[INFO] Paimon : Hive Connector : 2.3 ...................... SKIPPED
[INFO] Paimon : Hive Connector : 3.1 ...................... SKIPPED
[INFO] Paimon : Arrow ..................................... SKIPPED
[INFO] Paimon : Open API .................................. SKIPPED
[INFO] Paimon : Spark : 3.5 ............................... SKIPPED
[INFO] Paimon : Spark : 3.4 ............................... SKIPPED
[INFO] Paimon : Spark : 3.2 1.1-SNAPSHOT .................. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 13.998 s
[INFO] Finished at: 2025-01-20T15:49:45+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-jar-plugin:3.4.2:jar (default-jar) on project paimon-test-utils: The plugin org.apache.maven.plugins:maven-jar-plugin:3.4.2 requires Maven version 3.6.3 -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginIncompatibleException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :paimon-test-utils

```

### Anything else?

Apache Maven 3.5.3
openjdk 1.8.0_432

### Are you willing to submit a PR?

- [x] I'm willing to submit a PR!

--- a/README.md
+++ b/README.md
@@ -68,7 +68,7 @@ You can join the Paimon community on Slack. Paimon channel is in ASF Slack works
 
 ## Building
 
-JDK 8/11 is required for building the project. Maven version >=3.3.1.
+JDK 8/11 is required for building the project. Maven version >=3.6.3.
 
 - Run the `mvn clean install -DskipTests` command to build the project.
 - Run the `mvn spotless:apply` to format the project (both Java and Scala).

